wazo_webhookd/bin/sync_db.py

```diff
@@ -0,0 +1,150 @@
+
+from sqlalchemy import create_engine, or_
+from sqlalchemy.orm import scoped_session, sessionmaker
+from wazo_auth_client import Client as AuthClient
+from xivo import xivo_logging
+from xivo.chain_map import ChainMap
+from xivo.config_helper import read_config_file_hierarchy
+
+from wazo_webhookd.config import _DEFAULT_CONFIG, _load_key_file
+from wazo_webhookd.database.models import Subscription
+
+logger = logging.getLogger('wazo-webhookd-sync-db')
+
+
+def parse_cli_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        help="Log debug messages",
+    )
+    parser.add_argument(
+        '-q',
+        '--quiet',
+        action='store_true',
+        help='Only print warnings and errors',
+    )
+    parsed_args = parser.parse_args()
+    result = {'log_level': logging.INFO}
+    if parsed_args.quiet:
+        result['log_level'] = logging.WARNING
+    elif parsed_args.debug:
+        result['log_level'] = logging.DEBUG
+    return result
+
+
+def load_config():
+    file_config = read_config_file_hierarchy(ChainMap(_DEFAULT_CONFIG))
+    service_key = _load_key_file(ChainMap(file_config, _DEFAULT_CONFIG))
+    return ChainMap(service_key, file_config, _DEFAULT_CONFIG)
+
+
+@contextmanager
+def rw_session(Session) -> Generator[scoped_session, None, None]:
+    session = Session()
+    try:
+        yield session
+        session.commit()
+    except BaseException:
+        session.rollback()
+        raise
+    finally:
+        Session.remove()
+
+
+def main():
+    cli_args = parse_cli_args()
+    config = load_config()
+
+    xivo_logging.setup_logging('/dev/null', log_level=cli_args['log_level'])
+    xivo_logging.silence_loggers(['stevedore.extension'], logging.WARNING)
+
+    token = AuthClient(**config['auth']).token.new(expiration=300)['token']
+
+    del config['auth']['username']
+    del config['auth']['password']
+    tenants = AuthClient(token=token, **config['auth']).tenants.list()['items']
+    auth_tenants = {str(tenant['uuid']) for tenant in tenants}
+    logger.debug('Found %s wazo-auth tenants', len(auth_tenants))
+
+    users = AuthClient(token=token, **config['auth']).users.list(recurse=True)['items']
+    auth_users = {str(user['uuid']) for user in users}
+    logger.debug('Found %s wazo-auth users', len(auth_users))
+
+    engine = create_engine(
+        config['db_uri'],
+        pool_size=config['rest_api']['max_threads'],
+        pool_pre_ping=True,
+    )
+    Session = scoped_session(sessionmaker())
+    Session.configure(bind=engine)
+
+    with rw_session(Session) as session:
+        remove_tenants(session, auth_tenants)
+        remove_users(session, auth_users)
+
+
+def remove_tenants(session, auth_tenants):
+    webhookd_tenants = {
+        subscription.owner_tenant_uuid
+        for subscription in (
+            session.query(Subscription)
+            .filter(Subscription.owner_user_uuid != None)  # noqa
```

Comment:
This condition sound weird to me
It should not be:
```suggestion
        subscription.owner_tenant_uuid
        for subscription in (
            session.query(Subscription)
            .filter(Subscription.owner_tenant_uuid != None)  # noqa
```

Reply:
You're totally right, it's fixed now, thank you